server/storage/azure.go

```diff
@@ -0,0 +1,129 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+)
+
+type AzureStorage struct {
+	Storage
+	client          *azblob.Client
+	containerClient *container.Client
+	containerName   string
+	logger          *log.Logger
+}
+
+func getCredentials() (*azidentity.DefaultAzureCredential, error) {
+	return azidentity.NewDefaultAzureCredential(nil)
+}
+
+func NewAzureBlobStorage(ctx context.Context, storageAccountName string, containerName string, logger *log.Logger) (Storage, error) {
+	credentials, err := getCredentials()
+	if err != nil {
+		return nil, err
+	}
+
+	serviceUrl := fmt.Sprintf("https://%s.blob.core.windows.net", storageAccountName)
```

Comment:
i wonder if there's any compatible azure blob storage around, like there are for s3, that we could support as well if we build the service url in a non-hardcoded way. is there anything like this?

Reply:
I am not aware and didn't see anything in a quick search